File: C++/fig6.cpp
#include<iostream>
#include<fstream>
#include "NSFEM.h"
using namespace NSFEM;

main() {
    int i;
    double x,y;
    std::ofstream f;

    init("../cavity/node.txt",
         "../cavity/element.txt",
         "../cavity/stream_fix.txt",
         "../cavity/vortex_fix.txt",
         "../cavity/vortex_wall.txt");
    i = run(0.1, 0.01, 0.01);
    std::cout << i << std::endl;

    f.open("fig6uv.txt");
    for(i=0; i<element.nrows(); i++) {
        elem_center(x, y, i);
        f << x << ' ';
        f << y << ' ';
        f << velocity[i][0] << ' ';
        f << velocity[i][1] << '\n';
    }
    f.close();

    f.open("fig6ps.txt");
    for(i=0; i<node.nrows(); i++) {
        f << node[i][0] << ' ';
        f << node[i][1] << ' ';
        f << stream[i] << '\n';
    }
    f.close();

    f.open("fig6om.txt");
    for(i=0; i<node.nrows(); i++) {
        f << node[i][0] << ' ';
        f << node[i][1] << ' ';
        f << vortex[i] << '\n';
    }
    f.close();
}

File: fig6.py
import numpy as np
import matplotlib.pyplot as plt
from NSFEM import NSFEM
from contour import contour

ns = NSFEM('cavity/node.txt',
           'cavity/element.txt',
           'cavity/stream_fix.txt',
           'cavity/vortex_fix.txt',
           'cavity/vortex_wall.txt')

n = ns.run(0.1, 0.01, 0.01)
print(n)

plt.figure(figsize=(5,5))

x,y = ns.node.T

plt.subplot(221)
plt.axis('equal')
plt.axis([0,10,0,10])
ns.plot_velocity()
plt.plot([0,10,10,0,0], [0,0,10,10,0], 'k')
plt.box('off')
plt.xticks(())
plt.yticks(())

plt.subplot(222)
plt.axis('equal')
plt.axis([0,10,0,10])
contour(x, y, ns.stream, 20)
plt.plot([0,10,10,0,0], [0,0,10,10,0], 'k')
plt.box('off')
plt.xticks(())
plt.yticks(())

plt.subplot(223)
plt.axis('equal')
plt.axis([0,10,0,10])
contour(x, y, ns.vortex, 20)
plt.plot([0,10,10,0,0], [0,0,10,10,0], 'k')
plt.box('off')
plt.xticks(())
plt.yticks(())

plt.tight_layout()
plt.savefig('fig6.eps')
plt.show()
